crates/sui-types/src/temporary_store.rs

```diff
@@ -153,11 +149,12 @@ impl<'backing> TemporaryStore<'backing> {
     pub fn new(
         store: Arc<dyn BackingStore + Send + Sync + 'backing>,
         input_objects: InputObjects,
+        receiving_objects: Vec<ObjectRef>,
         tx_digest: TransactionDigest,
         protocol_config: &ProtocolConfig,
     ) -> Self {
         let mutable_inputs = input_objects.mutable_inputs();
-        let lamport_timestamp = input_objects.lamport_timestamp();
+        let lamport_timestamp = input_objects.lamport_timestamp(&receiving_objects);
```

Comment:
Updated this to now calculate the lamport timestamp based on both the input object and receiving argument versions.